py/frequenz/client/electricity_trading/_types.py

```diff
@@ -302,12 +301,35 @@ class DeliveryPeriod:
     duration: DeliveryDuration
     """The length of the delivery period."""
 
-    def __post_init__(self) -> None:
-        """Validate the parameters."""
-        if not isinstance(self.start, datetime):
-            raise ValueError("Start must be a datetime object.")
-        if not isinstance(self.duration, DeliveryDuration):
-            raise ValueError("Duration must be a DeliveryDuration object.")
+    def __init__(
+        self,
+        start: datetime,
+        duration: timedelta,
+    ) -> None:
+        """
+        Initialize the DeliveryPeriod object.
+
+        Args:
+            start: Start UTC timestamp represents the beginning of the delivery period.
+            duration: The length of the delivery period.
+
+        Raises:
+            ValueError: If the duration is not 5, 15, 30, or 60 minutes.
+        """
+        self.start = start
+        mapping = {
```

Comment:
I'd rather go for branching then for a map. Calculating the map and getting the value from some memory region is most likely slower then a simple if statement.

Reply:
https://docs.python.org/3/reference/compound_stmts.html#match

Reply:
I wasn't sure if it's possible to match over a timedelta object, but if possible that'd be the preferred way imo.

Reply:
Something around the snipped below should work
```python
minutes = start.total_seconds() / 60
match minutes:
    case 5:
        self.duration = DeliveryDuration.MINUTES_5
    case 15:
        self.duration = DeliveryDuration.MINUTES_15
    case 30:
        self.duration = DeliveryDuration.MINUTES_30
    case 60:
        self.duration = DeliveryDuration.MINUTES_60
    case _:
        self.duration = DeliveryDuration.UNSPECIFIED
```